Code/Framework/AzFramework/Platform/Linux/AzFramework/Application/Application_Linux.cpp

```diff
@@ -27,11 +27,14 @@ namespace AzFramework
         rlimit currentLimit;
         int get_limit_result = getrlimit(RLIMIT_NOFILE, &currentLimit);
         AZ_Warning("Application", get_limit_result == 0, "Unable to read current ulimit open file limits");
-        if ((get_limit_result == 0) && (currentLimit.rlim_cur < g_minimumOpenFileHandles || currentLimit.rlim_max < g_minimumOpenFileHandles))
+        // non-privileged systems may not update the maximum hard cap, only the current limit and it cannot exceed the max:
+        if ((get_limit_result == 0) && (currentLimit.rlim_cur < g_minimumOpenFileHandles) && ( currentLimit.rlim_max > 0))
         {
             rlimit newLimit;
-            newLimit.rlim_cur = g_minimumOpenFileHandles; // Soft Limit
-            newLimit.rlim_max = g_minimumOpenFileHandles; // Hard Limit
+            newLimit.rlim_cur = AZ::GetMin(g_minimumOpenFileHandles, currentLimit.rlim_max); // Soft Limit
```

Comment:
The `g_minimumOpenFileHandles` value is set to 65536 which is requires 3 bytes to represent and therefore can only be stored in an `int`. Should this value be 65535?

Reply:
I'm not sure what you mean, on linux we use constexpr rlim_t g_minimumOpenFileHandles = 65536L;
rlim_t is unsigned int,.  
```
constexpr rlim_t g_minimumOpenFileHandles = 65536L;
```
rlimit struct also uses rlim_t, and 65536 fits easily into unsigned int...

Reply:
I was thinking the limit was an unsigned short max.

Reply:
```
struct rlimit {
    rlim_t rlim_cur;  /* Soft limit */
    rlim_t rlim_max;  /* Hard limit (ceiling for rlim_cur) */
};
```
and rlim_t is unsigned int or even a unsigned int64 in linux.  From the docs:
```
The glibc getrlimit() and setrlimit() wrapper functions use a
       64-bit rlim_t data type, even on 32-bit platforms.  However, the
       rlim_t data type used in the getrlimit() and setrlimit() system
       calls is a (32-bit) unsigned long.  Furthermore, in Linux, the
       kernel represents resource limits on 32-bit platforms as unsigned
       long.  
```

Reply:
Ok, then I am guessing we set `65536L` as a soft max then.

Reply:
yes, we can't modify the hard max unless we have admin so this code should be setting the soft max to 65536, and leaving the hard max at wahtever the hard max was before.